src/storage/storage_structure/column.cpp

```diff
@@ -86,6 +86,9 @@ bool Column::isNull(common::offset_t nodeOffset, transaction::Transaction* trans
 
 void Column::setNull(common::offset_t nodeOffset) {
     nullColumn->setValue(nodeOffset);
+    auto walPageInfo = createWALVersionOfPageIfNecessaryForElement(nodeOffset, numElementsPerPage);
```

Comment:
hmm I think we only need these newly added lines when we are adding a new node, right? Can we change `CreateNode` to go through `Column::write`? So we can gradually get rid of `setNull` interface, whose functionality is overlapped with `write`.

Let's discuss this a bit more when you are online.